### PR TITLE
Added somax to input/output to restart file

### PR DESCRIPTION
**Pull request by babrodtk**
This adds SOMAX to the restart files. Will have a conflict with #1084, as they somewhat overlap.

### CODE REVIEW

**Comment by andlaus** [comment]
Looking at the code nothing obvious seems to be wrong, so no objections from my side; merge when you deem appropriate. (I haven't compiled/tested this, though.)

**Comment by andlaus** [comment]
jenkins run norne_ebos please

**Comment by babrodtk** [comment]
I had missed one small change in OPM/ewoms#156 which needs to be merged first.

**Comment by babrodtk** [comment]
Now that OPM/ewoms#156 has been merged I think this should compile fine. Could someone with the right privileges ask Jenkins to check this one?

**Comment by bska** [comment]
jenkins run spe polymer norne please

**Comment by babrodtk** [comment]
So the Travis CI build fails with this:
```
Error: Program threw an exception: Read of restart file: File does not contain SOMAX data
[...]
The following tests FAILED:
22 - compareRestartedSim_flow+SPE1CASE2_ACTNUM (Failed)
23 - compareRestartedSim_flow+SPE9_CP_SHORT (Failed)
```
which I think needs an update to the cases using restart files.

**Comment by atgeirr** [comment]
jenkins build this update_data please

**Comment by akva2** [comment]
These are restart tests. There are no reference files. A simulation is done for the full time, a restarted simulation is performed then the results are compared. No stored reference.

-------- Opprinnelig melding --------
Fra: "André R. Brodtkorb" <notifications@github.com>
Dato: 14.03.2017 11:45 (GMT+01:00)
Til: OPM/opm-simulators <opm-simulators@noreply.github.com>
Ko: Subscribed <subscribed@noreply.github.com>
Emne: Re: [OPM/opm-simulators] Added somax to input/output to restart file (#1087)


So the Travis CI build fails with this:

Error: Program threw an exception: Read of restart file: File does not contain SOMAX data
[...]
The following tests FAILED:
22 - compareRestartedSim_flow+SPE1CASE2_ACTNUM (Failed)
23 - compareRestartedSim_flow+SPE9_CP_SHORT (Failed)


which I think needs an update to the cases using restart files.

**Comment by atgeirr** [comment]
> These are restart tests. There are no reference files.

Thanks for the alert! I'll assume that the opm-data PR is unneeded, then, but keep it open in case I'm wrong.

**Comment by akva2** [comment]
@babrodtk you now have trigger rights for opm-simulators

**Comment by akva2** [comment]
the problem is that ebce7e0a only added the output to ebos not the legacy solver. this is flow legacy so it will fail on load because SOMAX is not in the restart file as expected.

**Comment by babrodtk** [comment]
@akva2 Yes, you are correct. This fails with flow_legacy, and its derivatives (in this case, flow_mpi).

**Comment by akva2** [comment]
the good news is ebos passes fine ;)

**Comment by atgeirr** [comment]
jenkins run spe norne norne_ebos please

**Comment by babrodtk** [comment]
Still failing, urk. 

```
The following tests FAILED:
	 17 - compareECLFiles_flow+SPE1CASE2 (Failed)
	 18 - compareECLFiles_flow_sequential+SPE1CASE1 (Failed)
	 19 - compareECLFiles_flow+SPE3CASE1 (Failed)
	 20 - compareECLFiles_flow+SPE9_CP_SHORT (Failed)
	 21 - compareECLFiles_flow_multisegment+2D_H__ (Failed)
Errors while running CTest
```

These tests fail with the following from what I can see:
```
Uncommon keywords for the two cases:
SOMAX
```

@akva2 Does it now make sense to to run the update data on opm-data?

**Comment by akva2** [comment]
yes, these are regression tests - compareECLFiles - we compare against a stored reference and now that has to be updated.

**Comment by babrodtk** [comment]
jenkins build this update_data please

**Comment by akva2** [comment]
as in you have to use the old PR for merge, the contents were updated but do not use 193, but rather 190

**Comment by akva2** [comment]
spe1case2 fails with additional errors. i saw that in a build i did as well, so i do not think this PR is the cause but we need to track that first.

**Comment by bska** [comment]
> spe1case2 fails with additional errors.

Any news on that?  It is starting to interfere with other work.

**Comment by akva2** [comment]
not from me.

**Comment by akva2** [comment]
bisection leads to https://github.com/OPM/opm-material/commit/e7fadc0884005d5c5685849e054678fd511b498c

**Comment by babrodtk** [comment]
jenkins build this update_data please

**Comment by atgeirr** [comment]
jenkins build this opm-data=195 please

File: opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp [hunk 1]
```diff
@@ -149,6 +149,17 @@ class SimulatorFullyImplicitBlackoilEbos
             // This is a restart, populate WellState and ReservoirState state objects from restart file
             output_writer_.initFromRestartFile(props_.phaseUsage(), grid(), state, prev_well_state, extra);
             initHydroCarbonState(state, props_.phaseUsage(), Opm::UgGridHelpers::numCells(grid()), has_disgas_, has_vapoil_);
+            {
+                const int num_cells = Opm::UgGridHelpers::numCells(grid());
+
+                typedef std::vector<double> VectorType;
+
+                VectorType& somax = state.getCellData( "SOMAX" );
```
Comment (atgeirr) [inline opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp]: This should have been a const reference.

**Comment by atgeirr** [comment]
jenkins build this update_data please

**Comment by atgeirr** [comment]
jenkins build this opm-data=196 please

**Comment by atgeirr** [comment]
Finally, Jenkins is green! Merging.

**Comment by atgeirr** [comment]
This PR makes the master incompatible with using old restart files. We intend to remedy this shortly, in the course of fixing the hysteresis restarting data.

**Comment by GitPaean** [comment]
I have no intention to revert this PR, while creating a reverting branch for my own immediate testing purpose.

**Pull request merged**